thoth/storages/graph/postgres.py

```diff
@@ -3846,26 +3847,35 @@ def create_python_cve_record(
         version_range: str,
         advisory: str,
         cve: str = None,
-    ) -> Tuple[CVE, bool]:
+    ) -> bool:
         """Store information about a CVE in the graph database for the given Python package."""
         package_name = self.normalize_python_package_name(package_name)
         package_version = self.normalize_python_package_version(package_version)
         with self._session_scope() as session, session.begin(subtransactions=True):
-            cve, _ = CVE.get_or_create(
-                session, cve_id=record_id, version_range=version_range, advisory=advisory, cve_name=cve
-            )
-            index = self._get_or_create_python_package_index(session, index_url, only_if_enabled=False)
-            entity, _ = PythonPackageVersionEntity.get_or_create(
-                session,
-                package_name=package_name,
-                package_version=package_version,
-                python_package_index_id=index.id,
-            )
-            _, existed = HasVulnerability.get_or_create(
-                session, cve_id=cve.id, python_package_version_entity_id=entity.id
-            )
+            if session.query(exists().where(CVE.cve_id == record_id)).scalar():
+                return True
+            else:
+                cve_instance = CVE(
+                    "advisory": advisory,
+                    "cve_name": cve,
+                    "cve_id": record_id,
+                    "version_range": version_range,
+                    "aggregated_at": datetime.now(),
+                )
+                session.add()
+
+                index = self._get_or_create_python_package_index(session, index_url, only_if_enabled=False)
+                entity, _ = PythonPackageVersionEntity.get_or_create(
+                    session,
+                    package_name=package_name,
+                    package_version=package_version,
+                    python_package_index_id=index.id,
+                )
+                HasVulnerability.get_or_create(
+                    session, cve_id=cve_instance.id, python_package_version_entity_id=entity.id
```

Comment:
@fridex My only concern is here whether that object `cve_instance` would have `id` parameter which is autoincremented, primary key, otherwise LGTM?

Reply:
Please do not make primary key out of it (it's better to have integer primary keys) - make it unique and let PosgreSQL create an index for it.

You can try and test it :)

PS: You can also try `cve=cve_instance`. Let me know which version works for you.

Reply:
Frido there are two things we have in `CVE`. First is `id` which is https://github.com/thoth-station/storages/blob/master/thoth/storages/graph/models.py#L368 and second `cve_id` which is https://github.com/thoth-station/storages/blob/master/thoth/storages/graph/models.py#L372. The first is `primary key, autoincremented` and the second one is `unique`.
Here, we are accessing `id`, not `cve_id`. Is my question clear now, I'm sorry for not explaining in the beginning :smile:

Reply:
So, the is that when we add a new record to the table, then that `id` is generated, right? Now, we want to access it, I'm not sure whether we can access it like in this way.

Reply:
Ah, right! Sorry, I overlooked it :)

Try with:

```
HasVulnerability.get_or_create(
                    session, cve_id=cve_instance.id, python_package_version_entity_id=entity.id
)
```

and with:

```
HasVulnerability.get_or_create(
                    session, cve=cve_instance, python_package_version_entity_id=entity.id
)
```

And ignore my comment regarding primary key - friday, too late... My brain is thinking about :beers: :smile:

Reply:
@fridex Thanks, I checked both of them and this `cve_id=cve_instance.id,` works.